src/main/kotlin/com/workos/fga/FgaApi.kt

```diff
@@ -0,0 +1,120 @@
+
+    return workos.get(
+      "/fga/v1/resources",
+      Resources::class.java,
+      RequestConfig.builder().params(params).build()
+    )
+  }
+
+  /** Create a new resource in the current environment. */
+  fun createResource(options: CreateResourceOptions): Resource {
+    return workos.post(
+      "/fga/v1/resources",
+      Resource::class.java,
+      RequestConfig.builder().data(options).build()
+    )
+  }
+
+  /** Update an existing resource. */
+  fun updateResource(resourceType: String, resourceId: String, options: UpdateResourceOptions): Resource {
+    return workos.put(
+      "/fga/v1/resources/$resourceType/$resourceId",
+      Resource::class.java,
+      RequestConfig.builder().data(options).build()
+    )
+  }
+
+  /** Delete a resource in the current environment. */
+  fun deleteResource(resourceType: String, resourceId: String) {
+    workos.delete("/fga/v1/resources/$resourceType/$resourceId")
+  }
+
+  /** Get a list of all existing warrants matching the criteria specified */
+  fun listWarrants(options: ListWarrantsOptions? = null): Warrants {
+    val params: Map<String, String> =
+      RequestConfig.toMap(options ?: ListWarrantsOptions()) as Map<String, String>
+
+    return workos.get(
+      "/fga/v1/warrants",
+      Warrants::class.java,
+      RequestConfig.builder().params(params).build()
+    )
+  }
+
+  /** Perform a warrant write (create/delete) in the current environment */
+  fun writeWarrant(options: WriteWarrantOptions): WriteWarrantResponse {
+    return workos.post(
+      "/fga/v1/warrants",
+      WriteWarrantResponse::class.java,
+      RequestConfig.builder().data(options).build()
+    )
+  }
+
+  /** Performs multiple warrant writes in the current environment */
+  fun batchWriteWarrants(options: List<WriteWarrantOptions>): WriteWarrantResponse {
+    return workos.post(
+      "/fga/v1/warrants",
+      WriteWarrantResponse::class.java,
+      RequestConfig.builder().data(options).build()
+    )
+  }
+
+  /** Perform a warrant check in the current environment */
+  fun check(options: CheckOptions, requestOptions: CheckRequestOptions? = null): CheckResponse {
+    val config = RequestConfig.builder()
+      .data(options)
+
+    if (requestOptions != null) {
+      config.headers(mapOf("Warrant-Token" to requestOptions.warrantToken))
+    }
+
+    return workos.post("/fga/v1/check", CheckResponse::class.java, config.build())
+  }
+
+  /** Perform a batch warrant check in the current environment */
+  fun checkBatch(options: CheckBatchOptions, requestOptions: CheckRequestOptions? = null): Array<CheckResponse> {
+    val config = RequestConfig.builder()
+      .data(options)
+
+    if (requestOptions != null) {
+      config.headers(mapOf("Warrant-Token" to requestOptions.warrantToken))
+    }
+
+    return workos.post("/fga/v1/check", Array<CheckResponse>::class.java, config.build())
+  }
+
+  /** Perform a query in the current environment */
+  fun query(options: QueryOptions): QueryResponse {
```

Comment:
We should include `CheckRequestOptions` on this method as well, in which case it might make sense to rename `CheckRequestOptions` to something more generic.

Reply:
I created `QueryRequestOptions` - this prevents us from having to change the interfaces if the two methods diverge a bit with these options and matches what we have in the other SDKs as well